discord/ext/commands/converter.py

```diff
@@ -524,6 +525,33 @@ async def convert(self, ctx, argument):
         except Exception as exc:
             raise BadInviteArgument() from exc
 
+class GuildConverter(IDConverter):
+    """Converts to a :class:`~discord.Guild`.
+
+    The lookup strategy is as follows (in order):
+
+    1. Lookup by ID.
+    2. Lookup by name. (There is no disambiguation for Guilds with multiple matching names).
+
+    .. versionadded:: 1.7
+    """
+
+    async def convert(self, ctx, argument):
+        match = self._get_id_match(argument)
+        result = None
+
+        if match is not None:
+            guild_id = int(match.group(1))
+            result = ctx.bot.get_guild(guild_id)
+
+        if result is None:
+            predicate = lambda g: g.name == argument
+            result = discord.utils.find(predicate, ctx.bot.guilds)
```

Comment:
`get` is probably better (it is faster than `find` for the single attribute case).